include/mango_v3.hpp

```diff
@@ -140,6 +142,39 @@ struct MangoAccount {
   uint8_t padding[5];
 };
 
+class MangoAccount {
+  // Creating a MangoAccount might involve non-trivial work like calling RPC
+  // methods, so we provide explict `MangoAccount::from()` overloaded set of
+  // functions and make the public constructor un-callable.
+  struct NotPubliclyConstructible {};
+
+ public:
+  MangoAccount(NotPubliclyConstructible /* npc */,
+               const MangoAccountInfo &accountInfo) noexcept
+      : accountInfo_(accountInfo) {}
+  // Use a prefetched accountInfo
+  static std::unique_ptr<MangoAccount> from(
+      const MangoAccountInfo &accountInfo) noexcept {
+    return std::make_unique<MangoAccount>(NotPubliclyConstructible{},
+                                          accountInfo);
+  }
+  // Fetch `accountInfo` from `endpoint` and decode it
+  static std::unique_ptr<MangoAccount> from(
+      const solana::PublicKey &pubKey,
+      const std::string &endpoint = MAINNET.endpoint) {
+    auto connection = solana::rpc::Connection(endpoint);
+    const auto &accountInfo =
+        connection.getAccountInfo<MangoAccountInfo>(pubKey.toBase58());
+    return std::make_unique<MangoAccount>(NotPubliclyConstructible{},
+                                          accountInfo);
+  }
+  i80f48 getLiquidationPrice() { return {}; }
```

Comment:
there is no method like that in mango unfortunately

Reply:
Great explanation, I didn't know this. I was trying to replicate the methods provided by the Typescript SDK in `MangoAccount.ts` but they don't explain these internal details. This method doesn't do anything for now but this explanation is great for future implementation.